pkg/provision/providers/qemu/pflash.go

```diff
@@ -70,19 +70,41 @@ func (p *provisioner) createPFlashImages(state *vm.State, nodeName string, pflas
 
 	if secureBootEnabled {
 		flashVarsPath := state.GetRelativePath(fmt.Sprintf("%s-flash_vars.fd", nodeName))
-		ovmfVars := "/usr/share/OVMF/OVMF_VARS_4M.fd"
 
-		for _, pflash := range pflashSpec {
-			for _, sourcePath := range pflash.SourcePaths {
-				if strings.Contains(sourcePath, "edk2-x86_64-secure-code.fd") {
-					// alpine
-					ovmfVars = "/usr/share/OVMF/OVMF_VARS.fd"
+		// Default search paths
+		ovmfVarsPathPrefixes := []string{
+			"/usr/share/ovmf",
+			"/usr/share/OVMF",
+		}
 
-					break
-				}
+		// Empty vars files
+		ovmfVarsFiles := []string{
+			"OVMF_VARS_4M.fd",
+			"OVMF_VARS.fd",
+		}
+
+		var ovmfVarsPaths []string
+
+		for _, p := range ovmfVarsPathPrefixes {
+			for _, f := range ovmfVarsFiles {
+				ovmfVarsPaths = append(ovmfVarsPaths, filepath.Join(p, f))
 			}
 		}
 
+		var ovmfVars string
+
+		for _, p := range ovmfVarsPaths {
```

Comment:
we can combine the loop above into this and breaking on the first existing file

Reply:
I've intentionally separated the two loops to allow for returning the considered paths in the error below, this is invaluable information when debugging a missing package, search path etc.

Reply:
i believe we can still do as the else condition for the `os.Stat` since we break on first file found, so end of loop means we can't find the file and directly return the error?

```go
var ovmfVarsPaths []string
var ovmfVars string

		for _, p := range ovmfVarsPathPrefixes {
                        found := false
			for _, f := range ovmfVarsFiles {
				ovmfVarsPaths = append(ovmfVarsPaths, filepath.Join(p, f))

                                if _, err := os.Stat(filepath.Join(p, f)); err == nil {
				ovmfVars = p
                                found = true
                      
				break
			}
                        if found {
                              break
                       }
			}
		}

if ovmfVars == "" {
			return nil, fmt.Errorf("failed to find OVMF vars in %q", ovmfVarsPaths)
		}
```

something around this (apologies for the bad formatting)

Reply:
(This change is not needed now that https://github.com/siderolabs/talos/pull/7403 is merged)